### ISSUE TITLE
Support normal pages as well

### ISSUE THREAD

**Issue by lcorneliussen** [nhoizey/jekyll-postfiles#18]
It would be great if this plugin wouldn't only support blog posts but also normal pages by putting images next to the permalink-defined location of static pages.

**Comment by nhoizey** [comment]
Indeed. Work is already planned for collections, but having any page support would be great.

**Comment by MoHKale** [comment]
@nhoizey has collections support been implemented yet? I'd be willing to implement it.

**Comment by nhoizey** [comment]
@MoHKale I didn't make any progress yet unfortunately, so that's great that you're willing to implement it, thanks! 👍

**Comment by MoHKale** [comment]
I'll get on it soon... may take a few weeks to get started (nearly christmas & all).

**Comment by nhoizey** [comment]
No problem, you're the lead on this! 🙂

**Comment by mohkale** [comment]
Finally got round to implementing collections support. Now all that's left should be normal pages... but I'm not really sure how to implement it or even whether there's a need to. Could you give me a use case of where unique grouping of assets for regular pages make sense?

**Comment by DirtyF** [comment]
@mohkale I've seen people, including me, group pages in a `_pages` folder and include it in their config, to get a cleaner root. But it looks like a special collection, looks like that kind of structure could be done now with #47 ?

```
_config.yml
_pages/
 - about/
   index.md
   team.jpg
- 404/
   index.md
   error.jpg
_posts/
_collections/
Gemfile
...
```

**Comment by mohkale** [comment]
@DirtyF so _pages is a collection? ~~I'm not sure whether it'll work with #47 because a postfile has to be at least one directory below the collection directory~~ Sorry. the indentation with about/ and 404/ was off and that confused me. Yeah, that seems like a good solution.

**Comment by mohkale** [comment]
@DirtyF though I've found a bug with #47 in that regard. It seems to me like it's outputting both the converted `.html` format files and the original unconverted markdown files. I'm not sure why. I'll look into it.

**Comment by mohkale** [comment]
@DirtyF I've got one issue with _pages. Is it already defined as a collection? I can't seem to add it to my collections hash in _config.yml and I also can't change the permalink for members of it.

Adding

```yaml
defaults:
  - scope:
      type: pages
    values:
      permalink: /:title:output_ext
```

gets me: `jekyll 3.8.6 | Error:  Invalid argument @ rb_sysopen - mohkale.io/bin/:title.html`. I truncated the path for simplicities purposes, but `bin` is my build directory for jekyll.

**Comment by DirtyF** [comment]
It's not a collection, just an [included folder](https://github.com/DirtyF/frank.taillandier.me/blob/master/_config.yml#L71).

**Comment by mohkale** [comment]
I'm not sure whether it'll work with #47 then. that patch specifically goes through files in collections.

**Comment by mohkale** [comment]
Just ran into a use case where this'd be convenient so it's probably worth considering how to implement it. We could just ask users to manually create a collection, `_pages` in this case and then use that with #47 ...  but I'm not a big a fan of the whole _collection naming scheme & that'd force paths for files in the root of your site into at least one subdirectory below the root of your source directory, so I'd appreciate an alternative.

We could add a config option where users can specify paths and each matching path is considered the root for a postfiles directory. I.E.

site:
```
- about-me/
  - banner.png
  - index.md
- faq/
  - styles/
    - main.css
  - index.md
- _config.yml
```

_config.yml:
```yaml
postfiles:
  include:
    - {about-me,faq}/
```

neither approach seems very intuitive.
